tst/regression/run_test.py

```diff
@@ -0,0 +1,130 @@
+# Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+# for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+# in the program are reserved by Triad National Security, LLC, and the U.S. Department
+# of Energy/National Nuclear Security Administration. The Government is granted for
+# itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+# license in this material to reproduce, prepare derivative works, distribute copies to
+# the public, perform publicly and display publicly, and to permit others to do so.
+#========================================================================================
+
+# Python modules
+import argparse
+import os
+import sys
+
+""" To prevent littering up imported folders with .pyc files"""
+sys.dont_write_bytecode = True
+
+# Parthenon modules
+import utils.test_case as tc 
+
+""" Split a path into folders """
+def SplitPathIntoFolders(path):
+    folders = []
+    while 1:
+        path, folder = os.path.split(path)
+        if folder != "":
+            folders.append(folder)
+        else:
+            if path != "":
+                folders.append(path)
+            break
+    return folders
+
+def checkRunScriptLocation(run_test_py_path):
+  
+    """ Check that run_test is in the correct folder """
+    folders = SplitPathIntoFolders(run_test_py_path)
+    if( os.path.join(folders[2], folders[1], folders[0]) != "parthenon/tst/regression" ): 
+        error_msg = "Cannot run run_test.py, it is not in the correct directory, must be "
+        error_msg += "kept in parthenon/tst/regression"
+        raise TestError(error_msg)
+
+    """ Check that test_suites folder exists """
+    if( not os.path.isdir(os.path.join(run_test_py_path,'test_suites')) ): 
+        raise TestError("Cannot run run_test.py, the test_suites folder is missing.")
+
+# Main function
+def main(**kwargs):
+
+    print("*****************************************************************")
+    print("Beginning Python regression testing script")
+    print("*****************************************************************")
+
+    run_test_py_path = os.path.dirname(os.path.realpath(__file__))
+    checkRunScriptLocation(run_test_py_path) 
+
+    print("Initializing Test Case")
+
+    test_case = tc.TestCase(run_test_py_path,**kwargs)
+
+    print("Cleaning output folder in test")
+
+    test_case.CleanOutputFolder()
+
+    test_case.Run()
+
+    test_result = test_case.Analyze()
+
+    if (test_result == True):
+        return 0
+    else:
+        raise TestError("Test " + test_case.test + " failed")
+
+
+# Exception for unexpected behavior by individual tests
+class TestError(RuntimeError):
+    pass
+
+# Execute main function
+if __name__ == '__main__':
+    help_msg = ('name of the test directory, relative to test_suites/, '
+               'excluding .py.')
+    parser = argparse.ArgumentParser()
```

Comment:
Might want to add a docstring to `ArgumentParser` itself to describe what the program does from the CLI. I.e., something along the lines of
```suggestion
    parser = argparse.ArgumentParser("Automatically run a Parthenon regression test.")
```